### PR TITLE
fix(VTreeView): Null check when building tree from children

### PR DESCRIPTION
**Pull request by JasonLandbridge**
## Description

When using VTreeView and the [item-children](https://vuetifyjs.com/en/api/v-treeview/#props-item-children) is null or undefined, then a `children.map` in `VTreeview.ts` will cause an error. 

image

## Motivation and Context
Adding a simple null/undefined check and then setting it to `[]` fixes this problem. 
[[Bug Report] VTreeNode: Null or Undefined children causing trouble with event bubbling #9376 (comment)](https://github.com/vuetifyjs/vuetify/issues/9376#issuecomment-572765655) 

## How Has This Been Tested?
Tested in my current project by adding the same code and this fixes the problem. Children which are undefined are not expandable. 

## Markup:

<details>

```vue
<template>
  <v-treeview :items="items" />
</template>

<script>
  export default {
    data: () => ({
      items: [
        {
          id: 'a',
          name: 'This one is  active with no exception',
        },
        {
          id: 'b',
          name: 'So it this one, you can see it as active',
          children: [
            {
              id: 'a',
              name: 'This one is  active with no exception',
            },
            {
              id: 'b',
              name: 'So it this one, you can see it as active',
              children: [],
            },
            {
              id: 'c',
              name: 'But this one is crashing during the build tree so its messing up',
              children: null,
            },
            {
              id: 'd',
              name: 'and so is tzhis one',
              children: undefined,
            },
          ],
        },
        {
          id: 'c',
          name: 'But this one is crashing during the build tree so its messing up',
          children: null,
        },
        {
          id: 'd',
          name: 'and so is tzhis one',
          children: undefined,
        },
      ],
    }),
  }
</script>
```
</details>

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [X] The PR title is no longer than 64 characters.
- [X] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [X] My code follows the code style of this project.
- [X] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by jacekkarczmarczyk** [comment]
1. this will check also for `0`, `false`, `''`, such cases i'd rather treat as an invalid value and throw the error. I'd suggest `const children = getObjectValueByPath(item, this.itemChildren) ?? []`
2. https://github.com/vuetifyjs/vuetify/issues/9376#issuecomment-572765655 - however I personally don't agree with it and think that `null` should also mean "no children"

**Comment by JasonLandbridge** [comment]
> 1. this will check also for `0`, `false`, `''`, such cases i'd rather treat as an invalid value and throw the error. I'd suggest `const children = getObjectValueByPath(item, this.itemChildren) ?? []`
> 2. [[Bug Report] VTreeNode: Null or Undefined children causing trouble with event bubbling #9376 (comment)](https://github.com/vuetifyjs/vuetify/issues/9376#issuecomment-572765655) - however I personally don't agree with it and think that `null` should also mean "no children"

Yes that is what I had initally but I wasn't sure if the null coalesce operator would work here.

Changed it!

File: packages/vuetify/src/components/VTreeview/VTreeview.ts [hunk 1]
```diff
@@ -201,7 +201,7 @@ export default mixins(
       for (let i = 0; i < items.length; i++) {
         const item = items[i]
         const key = getObjectValueByPath(item, this.itemKey)
-        const children = getObjectValueByPath(item, this.itemChildren, [])
+        const children = getObjectValueByPath(item, this.itemChildren, []) ?? []
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VTreeview/VTreeview.ts:204]:
```suggestion
        const children = getObjectValueByPath(item, this.itemChildren) ?? []
```

**Comment by JasonLandbridge** [comment]
@jacekkarczmarczyk , thanks for reviewing and helping out.! Is there something that needs to happen before this can be approved and merged?

**Comment by JasonLandbridge** [comment]
Hey @johnleider, Is there anything that needs to change in order for this pull request to get merged?

**Review by johnleider — APPROVED** [review]

**Comment by JasonLandbridge** [comment]
@johnleider, Thanks for taking a look! Is it possible this can be merged in Vuetify V2 since its very minor?

I'm asking because I now see the v3.0.0 milestone added.

**Comment by johnleider** [comment]
Good catch, just a mistake. Don't be alarmed 🔔

**Comment by JasonLandbridge** [comment]
> Good catch, just a mistake. Don't be alarmed 🔔

Awesome, thanks you very much!

**Review by jacekkarczmarczyk — COMMENTED** [review]
lgtm, I've added the playground

**Pull request merged**